core/src/main/scala/kafka/admin/TopicCommand.scala

```diff
@@ -313,42 +313,57 @@ object TopicCommand extends Logging {
     }
 
     override def describeTopic(opts: TopicCommandOptions): Unit = {
-      val topics = getTopics(opts.topic, opts.excludeInternalTopics)
-      ensureTopicExists(topics, opts.topic, !opts.ifExists)
+      val topicId = opts.topicId.map(Uuid.fromString).filter(_ != Uuid.ZERO_UUID)
+      // if topicId is provided and not zero, will use topicId regardless of topic name
```

Comment:
we would fail if both are provided?

Reply:
that's what I thought at the beginning, now I think it's better to use topicId and print a warning message if both are provided.

Reply:
I don't have an issue with this, but just curious why this was chosen over failing.